File: Rohit Singhal/LinkedLists/IntersectionofTwoLinkedLists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    
    public int length(ListNode head)
    {
        ListNode temp=head;
        int len=0;
        while(temp!=null)
        {
            len++;
            temp=temp.next;
        }
        return len;
        
    }
    
    
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        int len1=length(headA);
        int len2=length(headB);
        if(len1>len2)
        {
             for(int i=0;i<len1-len2;i++)
             {
                 headA=headA.next;
             }
        }
        else
        {
            for(int i=0;i<len2-len1;i++)
             {
                 headB=headB.next;
             }
        }
        while(headA!=null && headB!=null)
        {
            if(headA==headB)
            {
                return headA;
            }
            else
            {
                headA=headA.next;
                headB=headB.next;
            }
        }
        return null;
       
        
    }
}

File: Hashwanth/Linked Lists, Queues& Stacks/8_ Intersection of Two Linked Lists/Intersection of Two Linked Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        map<ListNode*,int> mp;
        ListNode *temp=headA;
        while(temp)
        {
            mp[temp]=1;
            temp=temp->next;
        }
        temp=headB;
        while(temp)
        {
            if(mp.find(temp)!=mp.end())
                return temp;
            temp=temp->next;
        }
        return NULL;
    }
 
    
};
